src/Symfony/Component/Validator/ConstraintViolationInterface.php

```diff
@@ -36,7 +36,7 @@ interface ConstraintViolationInterface
     /**
      * Returns the violation message.
      *
-     * @return string The violation message
+     * @return mixed The violation message
```

Comment:
i guess this shows you relied on implicit behavior :) we cant change it because of BC. So perhaps cast on read in its implementation?

But in general a message is a string, i tend to believe it should be passed as is.

Reply:
ro0NL - Thank you for reviewing this. I appreciate your input.

We've considered recasting on read. We can convert it back to a Markup object to make twig consider the html safe, but we can't store enough information in the violation to recreate the original Translatable Markup object that we use, so the translation happens upon constructing the violation rather than in the twig layer, which is not ideal.

Expanding the allowables types for the constructor is not a BC break, but expanding the return type of getMessage from string to mixed would be, but I would argue that the break is mitigated by the fact that 1) the return type is not technically a part of the interface (yet) 2) the return value in this PR is guaranteed to be castable to a string and 3) you would only get an object if you passed one in in the first place.

Reply:
> 1) the return type is not technically a part of the interface (yet)

that's wrong. Our BC policy covers the return type, even when only phpdoc is used to define it.

Reply:
let's keep it as `string` but alter the description instead, eg
`The violation message as a string or a stringable object`